apps/src/code-studio/components/progress/TeacherPanelProgressBubble.jsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import color from '@cdo/apps/util/color';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import {
+  DOT_SIZE,
+  DIAMOND_DOT_SIZE,
+  levelProgressStyle
+} from '@cdo/apps/templates/progress/progressStyles';
+import {LevelStatus} from '@cdo/apps/util/sharedConstants';
+
+/**
+ * A TeacherPanelProgressBubble represents progress for a specific level in the TeacherPanel. It can be a circle
+ * or a diamond. The fill and outline change depending on the level status.
+ */
+
+const styles = {
+  main: {
+    boxSizing: 'content-box',
+    fontFamily: '"Gotham 5r", sans-serif',
+    width: DOT_SIZE,
+    height: DOT_SIZE,
+    borderRadius: DOT_SIZE,
+    borderStyle: 'solid',
+    borderColor: color.lighter_gray,
+    fontSize: 16,
+    letterSpacing: -0.11,
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    transition:
+      'background-color .2s ease-out, border-color .2s ease-out, color .2s ease-out',
+    marginTop: 3,
+    marginBottom: 3
+  },
+  diamond: {
+    width: DIAMOND_DOT_SIZE,
+    height: DIAMOND_DOT_SIZE,
+    borderRadius: 4,
+    transform: 'rotate(45deg)',
+    marginTop: 6,
+    marginBottom: 6
+  },
+  contents: {
+    whiteSpace: 'nowrap',
+    lineHeight: '16px'
+  },
+  diamondContents: {
+    // undo the rotation from the parent
+    transform: 'rotate(-45deg)'
+  }
+};
+
+export class TeacherPanelProgressBubble extends React.Component {
```

Comment:
Any insights into whether we should consider consolidating this component with ProgressBubble in the future? Either way, launching with a separate component here seems totally fine since unifying them might be a bit of work and also risk regressions elsewhere.

Reply:
I went back and forth on this a bunch. The way the current ProgressBubble is has a lot of extra features to it that we don't want for this case. The ProgressBubble uses in other places needs to be able to link to other things and here we just wanted to display the bubble. It seemed like it was going to get really ugly to use the ProgressBubble here as it would add in a bunch of cases to that file. That being said I don't know if that's in keeping with what we have for best practices.

Reply:
my email response to this ended up on the PR below rather than inline with this comment.